completion/services.py

```diff
@@ -46,19 +58,44 @@ def get_completions(self, candidates):
         Parameters:
 
             candidates: collection of BlockKeys within the current course.
+            Note: Usage keys may not have the course run filled in for old mongo courses.
+            This method checks for completion records against a set of BlockKey candidates with the course run
+            filled in from self._course_key.
 
         Return value:
 
             dict[BlockKey] -> float: Mapping blocks to their completion value.
         """
-        completion_queryset = BlockCompletion.objects.filter(
-            user=self._user,
-            course_key=self._course_key,
-            block_key__in=candidates,
+        queryset = BlockCompletion.user_course_completion_queryset(self._user, self._course_key).filter(
+            block_key__in=candidates
         )
-        # pylint: disable=not-an-iterable
-        completions = {block.block_key: block.completion for block in completion_queryset}
-        for candidate in candidates:
+        completions = BlockCompletion.completion_by_block_key(queryset)
+        candidates_with_runs = [candidate.replace(course_key=self._course_key) for candidate in candidates]
+        for candidate in candidates_with_runs:
             if candidate not in completions:
                 completions[candidate] = 0.0
         return completions
+
+    def vertical_is_complete(self, item):
+        """
+        Calculates and returns whether a particular vertical is complete.
+        The logic in this method is temporary, and will go away once the
+        completion API is able to store a first-order notion of completeness
+        for parent blocks (right now it just stores completion for leaves-
+        problems, HTML, video, etc.).
```

Comment:
Is this no longer tested because we couldn't move over the tests that use the various xmodule factories?

Reply:
Yeah, kinda.  It's still tested in an integration test directory in edx-platform, so we've got that.  We could use mock to write tests here, too, but I might leave that for a future improvement.

Reply:
That sounds fine to me, especially because this code will go away when non-leaf completion is implemented.